### ISSUE TITLE
ERROR: f2d_register_rpc() => (null) (-22)

### ISSUE THREAD

**Issue by kschiess** [pthrasher/kckr#6]
kckr 2.0 here, on OS X, node 8.1.4.

When doing some operations on the watched directory, I get this error: 

~~~
2017-08-04 09:48 node[4989] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
2017-08-04 09:48 node[4989] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
events.js:182
      throw er; // Unhandled 'error' event
      ^

Error: Error watching file for changes: EMFILE
    at exports._errnoException (util.js:1024:11)
    at FSEvent.FSWatcher._handle.onchange (fs.js:1359:9)
~~~

kckr is launched here as: 

~~~
kckr -e "rsync --exclude .git -av . SOME_DEST" -r '.*\.rb$' .
~~~

Since your README says that kckr has no errors, this violates my expectations ;)

**Comment by pthrasher** [comment]
@kschiess hehe... apologies for violating your expectations. It's one of the worst kinds of violations. ;-)

I haven't worked on this in some time, as many much better tools have come around such as facebook/watchman

That said, I'm happy to look into this and see if I can figure out the problem as I'm sure watchman may not be good for all scenarios.

Thanks for the bug report! :-)

**Comment by pthrasher** [comment]
@kschiess I've swapped `fs` out for `graceful-fs` which should resolve this issue. Please open another issue if you have any other problems. :-)

**Comment by kschiess** [comment]
:D

**Comment by pthrasher** [comment]
@kschiess I originally published this as `2.1.0` -- but realized it was *potentially* a breaking change since I added a brand new dependency. So I released `2.1.1` which rolls this back, and then released `3.0.0` which is the version you want. :-)

**Comment by kschiess** [comment]
Thank you so much!
